pyproject.toml

```diff
@@ -0,0 +1,130 @@
+[build-system]
+requires = ["setuptools>=64", "setuptools_scm[toml]>=6.2", "wheel"]
+build-backend = "setuptools.build_meta"
+
+[project]
+name = "event-model"
+classifiers = [
+    "Development Status :: 3 - Alpha",
+    "License :: OSI Approved :: BSD License",
+    "Programming Language :: Python :: 3.8",
+    "Programming Language :: Python :: 3.9",
+    "Programming Language :: Python :: 3.10",
+    "Programming Language :: Python :: 3.11",
+]
+description = "One line description of your module"
+dependencies = [
+    "importlib-metadata",
+    "jsonschema",
+    "numpy",
+    "packaging"
+]
+dynamic = ["version"]
+license.file = "LICENSE"
+readme = "README.rst"
+requires-python = ">=3.8"
+
+[project.optional-dependencies]
+dev = [
+    "dask[array]",
+    "black",
+    "mypy",
+    "flake8",
+    "flake8-isort",
+    "Flake8-pyproject",
+    "pipdeptree",
+    "pre-commit",
+    "pydata-sphinx-theme>=0.12",
+    "pytest",
+    "pytest-cov",
+    "sphynx",
+    "sphinx-autobuild",
+    "sphinx-copybutton",
+    "sphinx-design",
+    "tox-direct",
+    "types-mock",
+
+    # These are dependencies of various sphinx extensions for documentation.
+    "ipython",
+    "matplotlib",
+    "numpydoc",
+    "sphinx-copybutton",
+]
+
+[project.scripts]
+# event_model = "event_model.__main__:main"
+
+[project.urls]
+GitHub = "https://github.com/bluesky/event-model"
+
+#[[project.authors]] # Further authors may be added by duplicating this section
+# email = "email@email.com"
+# name = "Firstname Lastname"
+
+
+[tool.mypy]
+# Ignore missing stubs in imported modules
+ignore_missing_imports = true
+
+
+[tool.isort]
+float_to_top = true
+profile = "black"
+
+[tool.flake8]
+extend-ignore = [
+    # See https://github.com/PyCQA/pycodestyle/issues/373
+    "E203",
+    # support typing.overload decorator
+    "F811",
+    # allow Annotated[typ, some_func("some string")]
+    "F722",
+]
+max-line-length = 115
+exclude = [".tox", ".venv", "venv"]
+
+
+[tool.pytest.ini_options]
+# Run pytest with all our checkers, and don't spam us with massive tracebacks on error
+addopts = """
+    --tb=native -vv
+    --cov=event_model --cov-report term --cov-report xml:cov.xml
+    """
+# https://iscinumpy.gitlab.io/post/bound-version-constraints/#watch-for-warnings
+filterwarnings = [
+    "error",
+    "ignore::DeprecationWarning"
+]
+# Doctest python code in docs, python code in src docstrings, test functions in tests
+testpaths = "event_model/tests"
+
+[tool.coverage.run]
+data_file = "/tmp/event_model.coverage"
```

Comment:
Good question, probably not unless it's in WSL. I'll change it to the local build directory just incase...

Reply:
This was put in because the default (in the root of the repo) was slow on NFS mounted directories. Putting the coverage file somewhere else made tests about 3x faster. Do we have users on windows? If so then we should just remove this line.

Reply:
The docs mention using windows, so unless that's out of date I'd be for removing this line to ensure support.